geth/mailservice/mailservice_api.go

```diff
@@ -0,0 +1,131 @@
+package mailservice
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv5"
+	"github.com/status-im/status-go/geth/log"
+)
+
+const (
+	// defaultWorkTime is a work time reported in messages sent to MailServer nodes.
+	defaultWorkTime = 5
+)
+
+var (
+	// ErrInvalidMailServerPeer is returned when it fails to parse enode from params.
+	ErrInvalidMailServerPeer = errors.New("invalid mailServerPeer value")
+	// ErrInvalidSymKeyID is returned when it fails to get a symmetric key.
+	ErrInvalidSymKeyID = errors.New("invalid symKeyID value")
+)
+
+// PublicAPI defines a MailServer public API.
+type PublicAPI struct {
+	provider ServiceProvider
+}
+
+// NewPublicAPI returns a new PublicAPI.
+func NewPublicAPI(provider ServiceProvider) *PublicAPI {
+	return &PublicAPI{provider}
+}
+
+// MessagesRequest is a payload send to a MailServer to get messages.
+type MessagesRequest struct {
+	// MailServerPeer is MailServer's enode address.
+	MailServerPeer string `json:"mailServerPeer"`
```

Comment:
Did you talked wit @oskarth about changing api struct?

Reply:
Yes.